rpm/SPECS/openresty-pcre.spec

```diff
@@ -0,0 +1,119 @@
+Name:       openresty-pcre
+Version:    8.39
+Release:    1%{?dist}
+Summary:    Perl-compatible regular expression library for OpenResty
+
+Group:      System Environment/Libraries
+
+License:    BSD
+URL:        http://www.pcre.org/
+Source0:    ftp://ftp.csx.cam.ac.uk/pub/software/programming/pcre/pcre-%{version}.tar.bz2
+
+# http://vcs.pcre.org/pcre/code/trunk/pcregrep.c?view=patch&r1=1658&r2=1657&pathrev=1658
+Patch0:     patch_1658.diff
+
+BuildRequires:  readline-devel
+BuildRequires:  autoconf
+BuildRequires:  automake
+BuildRequires:  coreutils
+BuildRequires:  gcc
+BuildRequires:  glibc-common
+BuildRequires:  libtool
+BuildRequires:  make
+# perl not used because config.h.generic is pregenerated
+# Tests:
+BuildRequires:  bash
+BuildRequires:  diffutils
+BuildRequires:  grep
+
+%define openresty_prefix /usr/local/openresty
+
+# Do not check for provides, our internals are not for others.
+AutoReqProv: no
+
+%description
+PCRE, Perl-compatible regular expression, library has its own native API, but
+a set of wrapper functions that are based on the POSIX API are also supplied
+in the libpcreposix library. Note that this just provides a POSIX calling
+interface to PCRE: the regular expressions themselves still follow Perl syntax
+and semantics.
+
+%package devel
+Summary:    Development files for %{name}
+Group:      Development/Libraries
+Requires:   %{name} = %{version}-%{release}
+
+%description devel
+Development files (Headers, libraries for dynamic linking, etc) for %{name}.
+
+%package static
+Summary:    Static library for %{name}
+Group:      Development/Libraries
+Requires:   %{name}-devel = %{version}-%{release}
+
+%description static
+Library for static linking for %{name}.
+
+%package tools
+Summary:    Auxiliary utilities for %{name}
+Group:      Development/Tools
+Requires:   %{name} = %{version}-%{release}
+
+%description tools
+Utilities demonstrating PCRE capabilities like pcregrep or pcretest.
+
+%prep
+%setup -q -n pcre-%{version}
+%patch0 -p2
+
+%build
+LDFLAGS="-Wl,-rpath,%{openresty_prefix}/%{_lib}" ./configure \
+  --prefix=%{openresty_prefix} \
```

Comment:
The PCRE library should be installed into the prefix `%{openresty_prefix}/pcre` (or better, just `%{pcre_prefix}` which should be defined previously in this spec file).